File: ENTORNOS DE DESARROLLO/ejemplosLenguajes-main/java/Ejemplo.java
/*
 * Java
 *  Multiparadigma: Imperativo, Orientado a Objetos, Funcional
 *  Tipado: Estático, Fuerte
 *  Compilado e Interpretado
 *  Creador: Sun Microsystems
 *  Año: 1995
 *  Extensiones: .java, .class
 *  IDE: Eclipse, NetBeans, IntelliJ IDEA
 *  Frameworks: Spring, Hibernate, Struts, JSF
 */

/*
 * NORMAS DE ESTILO
 * - Nombre de archivo: NombreClase.java
 * - Nombre de clase: NombreClase
 * - Nombres de variables siguieren notación camelCase
 * - Nombres de clases empiezan con mayúscula
 * - Nombres de métodos empiezan con minúscula
 * - Línea de 80 caracteres
 * - Sangría de 4 espacios
 * - Comentarios en línea y de varias líneas
 * - Comentarios en métodos y clases
 * - Uso de espacios en operadores y comas
 * - Llaves en la misma línea
 * - Uso de mayúsculas en constantes
 * - Uso de nombres descriptivos
 * - Uso de nombres en inglés
 */


/*
 * Comentario de varias líneas
 * Se crea con /* y se cierra con * /
 */

// Comentario de una línea
// Se crea con //

/*
 * El package es un contenedor de clases, se utiliza para organizar las clases
 * y evitar conflictos de nombres.
 */
 package java;

/*
 * La clase Ejemplo es la clase principal, es la clase que se ejecuta.
 * La clase debe tener el mismo nombre que el archivo.
 * Cada archivo .java puede tener una sola clase pública.
 */
public class Ejemplo {

    /*
     * El método main es el punto de entrada de la aplicación.
     * Debe ser público, estático y no devuelve nada.
     * Debe existir un solo método main por aplicación.
     */
    public static void main(String[] args) {
        // El tipado es estático, se debe declarar el tipo de dato de la variable
        int numero;
        // La asignación se realiza con el operador =
        numero = 5;
        // Se pueden declarar e inicializar en la misma línea
        int numero2 = 10;
        // Se pueden declarar varias variables del mismo tipo en la misma línea
        int numero3, numero4;
        // Se pueden asignar valores a varias variables en la misma línea
        numero3 = numero4 = 15; // Vigilad con estas instrucciones que dificultan la lectura

        // Ejemplo de tipado fuerte
        //numero = 20.0; // Error de compilación, no se puede asignar un double a un int
        
        System.out.println("Hola Mundo");
        
    }
}

File: ENTORNOS DE DESARROLLO/ejemplosLenguajes-main/python/ejemplo.py
'''
Python
    Multiparadigma: Imperativo, Orientado a Objetos, Funcional
    Tipado: Dinámico, Fuerte
    Interpretado
    Sintaxis clara y sencilla
    Creado por Guido van Rossum en 1991
    Extensiones: .py
    IDEs: PyCharm, Visual Studio Code, Sublime Text, Atom, Jupyter Notebook
    Librerías: Numpy, Pandas, Matplotlib, Scikit-learn, TensorFlow, Keras, PyTorch
    Frameworks: Django, Flask, Pyramid, Bottle, Tornado, Web2py
'''

'''
    NORMAS DE ESTILO
    - Sangría de 4 espacios
    - Máximo 79 caracteres por línea
    - Líneas en blanco para separar funciones y clases
    - Comentarios en líneas separadas
    - Comentarios en inglés
    - Nombres de variables en minúsculas, con guiones bajos para separar palabras
    - Nombres de funciones y clases en minúsculas, con guiones bajos para separar palabras
    - Nombres de clases en CamelCase
    - Nombres de constantes en mayúsculas, con guiones bajos para separar palabras
'''


'''
    Comentario de varias líneas
    Se crea con tres comillas simples o dobles
'''
# Comentario de una línea
# Se crea con el símbolo hashtag o parrilla

# El tipado dinámico significa que no es necesario declarar el tipo de dato de una variable
x = 5
print(x) # ?
x = "Hola Mundo!"
print(x) # ?
x = True
print(x) # ?
x = 20.5
print(x) # ?

# El tipado fuerte significa que no se pueden mezclar tipos de datos en una operación
edad = 36
nombre = "Jordi"
# mensaje = "Mi nombre es " + nombre + " y tengo " + edad + " años."
mensaje = "Mi nombre es " + nombre + " y tengo " + str(edad) + " años."

# Esta variable cumple con las normas de estilo (snake_case)
ejemplo_de_variable = 5

print("Hola Mundo!")